keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/ScalarDoubleTensor.java

```diff
@@ -353,6 +353,14 @@ public double standardDeviation() {
         throw new IllegalStateException("Cannot find the standard deviation of a scalar");
     }
 
+    @Override
+    public boolean equalsWithEps(DoubleTensor o, double epsilon) {
+        if (this == o) return true;
+        if (!this.hasSameShapeAs(o)) return false;
+        double difference = value - o.asFlatList().get(0);
```

Comment:
Since o has the same shape, it's ok to just `value - o.scalar()`. The asFlatList() will do a lot more than you need.